src/solver/mha/mha_solver_forward.cpp

```diff
@@ -109,45 +73,59 @@ bool MhaForward::IsApplicable([[maybe_unused]] const ExecutionContext& context,
         return false;
     }
 
-    const miopen::mha::MhaInputDescsForward& descsForward = problem.GetDescsForward();
-
-    auto [N, H, S, D] = miopen::tien<4>(descsForward.kDesc.GetLengths());
-
-    return !miopen::IsDisabled(MIOPEN_ENV(MIOPEN_DEBUG_ATTN_NAIVE_FWD)) //
-           && S <= std::numeric_limits<uint32_t>::max()                 //
-           && descsForward.kDesc.IsPacked()                             //
-           && descsForward.qDesc.IsPacked()                             //
-           && descsForward.vDesc.IsPacked()                             //
-           && descsForward.oDesc.IsPacked()                             //
-           && descsForward.mDesc.IsPacked()                             //
-           && descsForward.zInvDesc.IsPacked()                          //
-           && MIOPEN_USE_GEMM;
+    const miopen::mha::MhaInputDescsForward& descsFwd = problem.GetDescsForward();
+
+    auto [N, H, S, D] = miopen::tien<4>(descsFwd.kDesc.GetLengths());
+
+    return MIOPEN_USE_GEMM                                                          //
```

Comment:
Is this a macro? Perhaps we should cast it to `bool` ?

Reply:
Probably.
It is extensively used in `#if MIOPEN_USE_GEMM`, so it must be implicitly castable to bool in all cases.